crates/neon-sys/native/src/neon_threadsafe_cb.h

```diff
@@ -0,0 +1,99 @@
+#ifndef NEON_THREADSAFE_CALLBACK_H_
+#define NEON_THREADSAFE_CALLBACK_H_
+
+#include <uv.h>
+#include "neon.h"
+#include "v8.h"
+#include <mutex>
+
+namespace neon {
+
+    class ThreadSafeCallback {
+    public:
+        ThreadSafeCallback(v8::Isolate *isolate,
+                           v8::Local<v8::Value> self,
+                           v8::Local<v8::Function> callback): isolate_(isolate), close_(false)
+        {
+            async_.data = this;
+            uv_async_init(uv_default_loop(), &async_, async_callback);
+            // Save the this argument and the callback to be invoked.
+            self_.Reset(isolate, self);
+            callback_.Reset(isolate, callback);
+            // Save the context (aka realm) to be used when invoking the callback.
+            context_.Reset(isolate, isolate->GetCurrentContext());
+        }
+
+        void call(void *rust_callback, Neon_ThreadSafeCallbackHandler handler) {
```

Comment:
I would recommend we call this `schedule` for clarity and consistency with the public API we're implementing.